### PR TITLE
feat: remove RwLock on entire ReadBuffer

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/669

# Rationale
Previously the ReadBuffer had no synchronization internally and thus we synchronized it with a giant lock

As of https://github.com/influxdata/influxdb_iox/pull/710 the ReadBuffer now handles synchronization internal to itself, so there is no need to lock the entire thing.

# Changes:
Remove lock + update the code accordingly (aka do what the compiler tells me)

# Commentary
Rust is awesome: I am much less terrified of this kind of change than I would be in C++ codebases

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: server/src/db.rs [hunk 1]
```diff
@@ -83,9 +83,7 @@ pub struct Db {
     #[serde(skip)]
     /// The read buffer holds chunk data in an in-memory optimized
     /// format.
-    ///
-    /// TODO: finer grained locking see ticket https://github.com/influxdata/influxdb_iox/issues/669
-    pub read_buffer: Arc<RwLock<ReadBufferDb>>,
```
Comment (alamb) [inline server/src/db.rs:88]: This is the actual change of this PR, and everything else is to make sure it compiles

File: read_buffer/src/lib.rs [hunk 2]
```diff
@@ -103,7 +103,7 @@ impl Database {
     /// chunk. If the `Table` or `Chunk` does not exist they will be created,
     /// otherwise relevant structures will be updated.
     pub fn upsert_partition(
-        &mut self,
```
Comment (alamb) [inline read_buffer/src/lib.rs:106]: These functions no longer need a &mut self as they synchronize themselves internally.

**Review by pauldix — APPROVED** [review]
Nice, remove all the locks! Well, the unnecessary ones anyway.

**Comment by alamb** [comment]
FYI @e-dard

**Pull request merged**